src/FeedIo/Reader/Fixer/HttpLastModified.php

```diff
@@ -0,0 +1,35 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of the feed-io package.
+ *
+ * (c) Alexandre Debril <alex.debril@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FeedIo\Reader\Fixer;
+
+use FeedIo\Reader\FixerAbstract;
+use FeedIo\Reader\Result;
+
+class HttpLastModified extends FixerAbstract
+{
+
+    /**
+     * @param Result $result
+     * @return FixerAbstract
+     */
+    public function correct(Result $result): FixerAbstract
+    {
+        $feed = $result->getFeed();
+        $response = $result->getResponse();
+
+        if ($response->getLastModified() instanceof \DateTime) {
```

Comment:
be careful, this will set the wrong value in some cases.
Take https://www.php.net/feed.atom:
- the feed was actually updated at 2019-12-20T14:48:33+00:00
- the last-modified header sent by the server is `last-modified: Wed, 08 Jan 2020 12:10:09 GMT`
so setting the feed's lastModified attribute without checking if it's empty before would lead to wrong value in it

Reply:
Good catch!

https://github.com/alexdebril/feed-io/blob/6e895196e1f5391a9f1412bc810064b25e464a5b/tests/samples/rss/sample-rss.xml#L9-L10

1. If one of those elements is present parser will set it as lastModified
2. If lastModified empty: check if last-modified header is present 
3. If lastModified empty: check feed items for pubDate

That's the right way? So I need to check that with httpLastModified fixer. Will do ;)

Reply:
yes, that's the good process :)
about the duplication: you could use a ResultMockBuilder (or Factory, it's quite the same) instead of `getResultMock()` methods .

Reply:
Done ;)